### PR TITLE
Align documentation for C:\ProgramData\Git\config

### PR DESCRIPTION
**Pull request by asheiduk**
Fixed the confusing and inconsistent documentation for this configuration file
by

 - aligning it with the already existing upstream predecence rules
 - removing the XP-hack using ``$ALLUSERSPROFILE/Application Data` instead
   of `$PROGRAMDATA` both in code and in docu.

Signed-off-by: Andreas Heiduk <asheiduk@gmail.com>

### CODE REVIEW

File: Documentation/git-config.txt [hunk 1]
```diff
@@ -235,8 +235,12 @@ FILES
 If not set explicitly with `--file`, there are four files where
 'git config' will search for configuration options:
 
+$PROGRAMDATA/Git/config::
+	System-wide configuration file for *all* Git-related software running
```
Comment (nalla) [inline Documentation/git-config.txt]: If you read on you will find a part where it is described that `--system` will target the _system-wide configuration_. If you leave the new text as-is, how does a user know which one `--system` targets?

Comment (dscho) [inline Documentation/git-config.txt]: Please note that this is Windows-only, too, so starting the description with `(Windows-only)` may be a very good idea here.

Together with @nalla's insightful observation, maybe something like this?

> (Windows-only): System-wide configuration file shared with other Git implementations.

File: Documentation/git-config.txt [hunk 2]
```diff
@@ -235,8 +235,12 @@ FILES
...
+	on the machine.	Typically `$PROGRAMDATA` points to `C:\ProgramData`.
+
 $(prefix)/etc/gitconfig::
-	System-wide configuration file.
+	System-wide configuration file for this installation of Git for Windows.
```
Comment (dscho) [inline Documentation/git-config.txt]: As we want to stay as close to the Git project (i.e. the platform-independent upstream project), it should be a good idea to keep the original intact, and only add something like:

> (Windows-only) this file is *not* intended to be shared with other Git implementations (such as JGit, libgit2, etc).

File: Documentation/git-config.txt [hunk 3]
```diff
@@ -253,14 +257,6 @@ $XDG_CONFIG_HOME/git/config::
...
 	Repository specific configuration file.
 
-On Windows, as there is no central `/etc/` directory, there is yet another
-config file (located at `$PROGRAMDATA/Git/config`), intended to contain
-settings for *all* Git-related software running on the machine. Consequently,
-this config file takes an even lower precedence than the
-`$(prefix)/etc/gitconfig` file. Typically `$PROGRAMDATA` points to
-`C:\ProgramData` (on Windows XP the equivalent in `$ALLUSERSPROFILE` is used,
-i.e. `C:\Documents and Settings\All Users\Application Data\Git\config`).
-
```
Comment (dscho) [inline Documentation/git-config.txt]: This explanation would need to be kept in one form or another, but I agree that with the additions above, it should be reworded because it now adds depth to the information above (as opposed to adding new information that has not been hinted at before).

File: compat/mingw.c [hunk 4]
```diff
@@ -3304,13 +3304,8 @@ const char *program_data_config(void)
 
 	if (!initialized) {
 		const char *env = mingw_getenv("PROGRAMDATA");
-		const char *extra = "";
-		if (!env) {
-			env = mingw_getenv("ALLUSERSPROFILE");
-			extra = "/Application Data";
-		}
```
Comment (dscho) [inline compat/mingw.c]: This needs to be in a separate commit, as we do not want to mingle different purposes in a single commit. One commit should remove the XP support, one should enhance the documentation.

**Comment by dscho** [comment]
I like the idea of this PR very much.

Please split the changes into two commits, though.

While at it, please adjust the commit messages to align with the rest of our commit messages by using the imperative instead of the past tense.

Also: the commit removing XP support should *at least* mention that v2.10.0 was the last Git for Windows version to support XP, and it would be really good to accompany that with a date, with a summary of the rationale (XP is EOL, but please be more verbose than that), and I guess a link to https://git-for-windows.github.io/requirements.html could not hurt.

Thanks!

**Comment by asheiduk** [comment]
@dscho Regarding the "extract the XP part" commit: Do you really want all the things after your "Also:" to be mentioned in the commit message?

File: Documentation/git-config.txt (hunk 1 above)
Comment (asheiduk) [inline Documentation/git-config.txt]: @nalla The "reading" part mentions that files are read in top-down order but for writing to --global (where already two files exist) the priority is bottom-up. I just assumed the same logic for two --system files.

File: Documentation/git-config.txt (hunk 3 above)
Comment (asheiduk) [inline Documentation/git-config.txt]: The upstream text puts all details about individual files into the list above. The XDG user-config and the standard ~/.gitconfig demonstrate this quite well. After the list only generic stuff about read / write priorities are described. Putting one block with in-depth information about one file between the list and the text is exactly the thing I'd like to avoid. So if the text must be kept, I'd put it into the appropriate list item.

File: compat/mingw.c (hunk 4 above)
Comment (asheiduk) [inline compat/mingw.c]: I admit, my only intention was to avoid dragging around the outdated description.

File: Documentation/git-config.txt (hunk 1 above)
Comment (dscho) [inline Documentation/git-config.txt]: For writing, there is no order: you have to specify *which* config you want to write to (default is the repository config).

File: Documentation/git-config.txt (hunk 3 above)
Comment (dscho) [inline Documentation/git-config.txt]: Fair enough. But the information cannot just go away.

**Comment by dscho** [comment]
> Regarding the "extract the XP part" commit: Do you really want all the things after your "Also:" to be mentioned in the commit message?

I would really like all of those, yes. I am a heavy user of the `git log`, especially when bug hunting (which is what I spend most of my time on, and I typically have to mindshare between multiple tickets, so every bit of information that is stored in the `git log` helps me).

File: Documentation/git-config.txt (hunk 3 above)
Comment (asheiduk) [inline Documentation/git-config.txt]: @dscho : The following info was there previously:

- ALLUSERPROFILE - not necessary any more (I've split the patch as you requested)
- PROGRAMMDATA - is there
- difference between both system configs - is there
- precedence rules - already covered by the (unchanged) upstream text

 (scratching head) What's missing?

Comment (asheiduk) [inline Documentation/git-config.txt]: @dscho Regarding the "not meddling" part: I clarified this yesterday in 58daf7e. Did you take into account these changes in your last reply? I'm unsure about the github tooling in this point...

Comment (dscho) [inline Documentation/git-config.txt]: Let me have a fresh look...

You're absolutely correct, I missed those updates. Thanks for the prod. The current version is awesome, thank you so much, will merge!
